src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -92,6 +92,129 @@ def _get_symbol_table(node):
     return None
 
 
+def check_bound(array, dim, index, operator):
```

Comment:
Please prefix this name with an underscore (as it is an internal utility) and make it more descriptive. In fact, could this routine be moved inside the check_range utility?

Reply:
In fact, is there scope to share this checking functionality with the routine that you've added to the backend?

Reply:
I've renamed to `_check_bound_is_full_extent` which reflects the change made to `check_range`. It can't be moved into `check_range` as it is also used in isolation for tests in `fparser2_test.py` e.g. if testing upper bound in `a(2:)`

Reply:
In theory there could be re-use between the front and back end code but that would involve me modifying the back-end function. However, the back-end code is permanent and the front end code will be deleted once the restriction of `:` is removed from the front-end `where` code, so I propose we leave it as it is.

Reply:
OK. Please add a comment saying this to the frontend code (if there isn't one already). Do we have an Issue for supporting bounds in a WHERE statement?

Reply:
I've created issue #717 and added text to the three routines relevant routines.

Reply:
Thanks.